docs/site/Troubleshooting.md

```diff
@@ -0,0 +1,68 @@
+---
+lang: en
+title: 'Troubleshooting'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Troubleshooting.html
+---
+
+## Common Troubleshooting Tips
+
+- You can get more information when the application in a debug mode. That is,
+  instead of starting the application using `npm start`, use this:
+
+  ```sh
+  DEBUG=loopback:* npm start
+  ```
+
+See more details on setting debug strings:
+https://loopback.io/doc/en/lb4/Setting-debug-strings.html
+
+## General
+
+### The latest change is not reflected in the application behavior
+
+For example, the error keeps complaining about a Controller class I just
+deleted.
+
+#### What does that mean?
+
+The transpiled JavaScript files probably are not in sync with the TypeScript
+file.
+
+#### How to fix?
+
+Make sure `npm run build` is being run after every change.
+
+If you have done so but still getting this error, it is possible that the
+transpiled JavaScript files did not get updated properly. Run `npm run clean` to
+delete all the transpiled files. By doing so, it forces the build to generate
+the transpiled JS files next time when you start the application using
+`npm start` or call `npm run build`.
+
+## Binding related
+
+### Binding key not bound error
+
+You get a similar error as below:
+
+```
+500 Error: The key 'services.hasher' is not bound to any value in context application
+```
+
+#### What does that mean?
+
+It means the binding key is not bound to a value.
+
+#### How to fix?
+
+Set the value for the binding key in the application. i.e. Go to your
+application, typically it is in `src/application.ts`,
+
+```ts
+// if your binding key is meant to bind to a class
+this.bind(SOME_BINDING_KEY).toClass(AClass);
+```
+
+See more details in
+[Binding documentation page](https://loopback.io/doc/en/lb4/Binding.html).
```

Comment:
1. We probably should mention how to add the application or module to `sandbox` so that it can be linked to latest LB code from git.

2. Do we have some instructions about running an application or mocha test in debugger with VSCode?

Reply:
+1

Personally, I am fine to land this PR as-is and iterate on the content in follow-up pull requests.

> We probably should mention how to add the application or module to sandbox so that it can be linked to latest LB code from git.

See the discussion in https://github.com/strongloop/loopback-next/issues/514.

Adding an app to `sandbox` is fine when the app has it's own git repo. I am not sure if this approach works for monorepos too? Use case to consider: how to run https://github.com/strongloop/loopback4-example-shopping with unreleased version of LB4.

A possible solution: 
1. Setup a local npm proxy registry (e.g. using https://verdaccio.org)
2. Publish from loopback-next to that private registry
3. In the debugged project, install dependencies from the private registry (note that proxy registries fall back to the public registry for packages that were not published privately).

> Do we have some instructions about running an application or mocha test in debugger with VSCode?

IDK. It would be good to include those instructions in https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/project/templates/DEVELOPING.md too.